template/src/ExamplePanel.tsx

```diff
@@ -23,7 +23,7 @@ function ExamplePanel({ context }: { context: PanelExtensionContext }): JSX.Elem
       // rendering before the next render call, studio shows a notification to the user that your panel is delayed.
       //
       // Set the done callback into a state variable to trigger a re-render.
-      setRenderDone(() => done);
+      setRenderDone(done);
```

Comment:
We don't need a closure here as the panel extension adapter creates a new function for every render. It turned out that removing this closure fixes the memory leak I was encountering.

Reply:
This change is wrong — I think you might be misreading the existing code. It is not `() => done()` but `() => done`. The reason is since useState supports a function argument `(oldState) => newState`, it assumes the given `done` function is supposed to be called and calls it immediately rather than updating the state variable. See docs: https://react.dev/reference/react/useState#im-trying-to-set-state-to-a-function-but-it-gets-called-instead